.github/workflows/generate-documentation.yml

```diff
@@ -0,0 +1,28 @@
+name: Generate documentation
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: 
+      - master
+  push:
+    branches: 
+      - master
```

Comment:
:warning: I suggest to ignore some path related to documentation to avoid extra build and extra GitHub Pages push in the future.
https://docs.github.com/en/enterprise-server@2.22/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

Reply:
fixed with 46a28fe